File: src/PerfTest/StructClass/Vector2S.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace PerfTest.StructClass
{
    public struct Vector2S
    {
        public int X;

        public int Y;

        public static Vector2S CreateRandom()
        {
            return new Vector2S
            {
                X = RandomService.Next(10),
                Y = RandomService.Next(10)
            };
        }

        public static IEnumerable<Vector2S> CreateN(int n)
        {
            for (var i = 0; i < n; ++i)
            {
                yield return CreateRandom();
            }
        }
    }
}

File: perf_test/StructClass/Vector2S.cpp
#include "Vector2S.h"
#include <stdlib.h>

using namespace StructClass;

StructClass::Vector2S::Vector2S(int x, int y)
{
	X = x;
	Y = y;
}

Vector2S Vector2S::CreateRandom()
{
	return Vector2S(abs(rand() % 10), abs(rand() % 10));
}

std::vector<Vector2S> StructClass::Vector2S::CreateN(size_t size)
{
	std::vector<Vector2S> vec(size);
	for (auto& v : vec)
	{
		v = CreateRandom();
	}
	return vec;
}
